src/esm_parser/provenance.py

```diff
@@ -37,9 +37,24 @@
 
 import copy
 
+from loguru import logger
+from ruamel.yaml import YAML
+
 import esm_parser
+import esm_tools
 from esm_calendar import Date
 
+CATEGORY_HIERARCHY = [
+    "defaults",  # Lowest in the hierarchy
+    "other_software",
+    "machines",
+    "components",
+    "setups",
+    "couplings",
+    "runscript",
+    "none",  # Highest in the hierarchy (no category means it is a change
```

Comment:
Here, and below, you can use None directly. I don't think you need a string. If you want a name, something like "set-by-function" (and maybe some logic to add the function name?)

Reply:
I have changed it with `backend`. The logic to add the function name already exist in the different methods of the `Provenance` class, but this does not belong to the `category` key of the provenance, but on the `modified_by` key.